src/model/CoilCoolingWater.cpp

```diff
@@ -313,9 +313,22 @@ namespace model {
                     << " cannot be connected directly to an AirLoopHVAC when it's part of a parent CoilSystemCoolingWaterHeatExchangerAssisted. "
                        "Please call CoilSystemCoolingWaterHeatExchangerAssisted::addToNode instead");
       } else {
-
-        success = WaterToAirComponent_Impl::addToNode(node);
+        // If trying to add to a PlantLoop, this will end up calling CoilCoolingWater_Impl::removePlantLoop which also removes the existing ControllerWaterCoil
+        auto t_plantLoop = node.plantLoop();
         auto t_containingZoneHVACComponent = containingZoneHVACComponent();
+        if (t_plantLoop && !t_containingZoneHVACComponent) {
+          if (!t_plantLoop->demandComponent(node.handle())) {
+            return false;
+          }
+
+          WaterToAirComponent_Impl::removeFromPlantLoop();  // NOLINT(bugprone-parent-virtual-call)
+
+          success =
+            HVACComponent_Impl::addToNode(node, t_plantLoop->demandInletNode(), t_plantLoop->demandOutletNode(), waterInletPort(), waterOutletPort());
+
+        } else {
+          success = WaterToAirComponent_Impl::addToNode(node);
+        }
```

Comment:
Found that out during testing... It was always removed when adding to a PlantLoop, so I had to check for that...
There are potential side effects here if the user had set the Actuator node to a node on a former plantloop though, maybe we should reset this?

Reply:
(there's no resetActuatorNode though)

Reply:
@kbenne